pkg/compactor/blocks_cleaner.go

```diff
@@ -628,12 +637,20 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userLogger log.Logger, us
 		level.Info(userLogger).Log("msg", "finish cleaning partial blocks", "duration", time.Since(begin), "duration_ms", time.Since(begin).Milliseconds())
 	}
 
-	// Upload the updated index to the storage.
-	begin = time.Now()
-	if err := bucketindex.WriteIndex(ctx, c.bucketClient, userID, c.cfgProvider, idx); err != nil {
-		return err
+	if idx.IsEmpty() && len(partials) == 0 {
```

Comment:
If the bucket index is empty but we have partial blocks, we are going to still write the index. Are we waiting for cleaner to clean up partial blocks and eventually we will delete the empty bucket index?

Reply:
That's right. If there are partial blocks, we need cleaner to delete those first. Bucket index file should be deleted if it is the only file in bucket store for a tenant.